src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SPECIAL_CHAR = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
+    private StringBuilder getFileStat(String fromFileName) {
```

Comment:
this method is too complex and large

you should have 3 separate methods:

1. Read the file data and return it (`String` or `String[]` should be fine)
2. Calculate the report based on the data you read in the first method
3. Write the report to a file

Currently this method reads the data and calculate the report at the same time